test/integration/test_dataplane.go

```diff
@@ -33,6 +33,33 @@ const (
 	tickTime = 250 * time.Millisecond
 )
 
+func updateAndVerifyServiceType(t *testing.T, dataplaneName types.NamespacedName, clients testutils.K8sClients, dataplane *operatorv1beta1.DataPlane, dataplaneIngressService corev1.Service, serviceType corev1.ServiceType) {
+	t.Helper()
+
+	t.Logf("updating dataplane spec with proxy service type of %s", serviceType)
+	require.Eventually(t,
+		testutils.DataPlaneUpdateEventually(t, GetCtx(), dataplaneName, clients, func(dp *operatorv1beta1.DataPlane) {
```

Comment:
Just to make this function no rely on `GetCtx()` we can pass the `context.Context` to it in an argument.

This will not only make this easier to reason about but also easier to refactor (e.g. when we'd like to move this function elsewhere).

```suggestion
		testutils.DataPlaneUpdateEventually(t, ctx, dataplaneName, clients, func(dp *operatorv1beta1.DataPlane) {
```